src/lib/models/index.ts

```diff
@@ -22,6 +26,47 @@ if (config.dbURL) {
   sequelize = new Sequelize(dbconfig.database, dbconfig.username, dbconfig.password, dbconfig)
 }
 
+const umzug = new Umzug({
+  migrations: {
+    path: path.resolve(__dirname, '..', 'migrations'),
+    params: [
+      sequelize.getQueryInterface(),
+      SequelizeTypes
+    ]
+  },
+  // Required wrapper function required to prevent winstion issue
+  // https://github.com/winstonjs/winston/issues/1577
+  logging: message => {
+    logger.info(message)
+  },
+  storage: 'sequelize',
+  storageOptions: {
+    sequelize: sequelize
+  }
+})
+
+// Semicolon is required to prevent `TypeError: (intermediate value) is not a function`-error
+;(async () => {
+  // checks migrations and run them if they are not already applied
+  // exit in case of unsuccessful migrations
+  await umzug.up().catch(error => {
+    logger.error(error)
+    logger.error('Database migration failed. Exciting…')
+    process.exit(1)
+  })
+  dbMigrationsComplete = true
+  logger.info('All migrations performed successfully')
+})()
+
+export async function waitForDBReady (): Promise<boolean> {
+  while (true) {
```

Comment:
I really don't like waiting for a global variable that's set by an anonymous, self-executing async function. Can't we move the whole `await umzug.up().catch(...` codeblock directly into this function ?

Reply:
Good point… Not sure why I didn't come up with this one :D